File: sorting/easy/selection.rb
def selection a
    for i in 0...a.length-1
        min = a[i]
        min_index = i
        for j in i+1...a.length
            if a[j] < min
                min = a[j]
                min_index = j
            end
        end
        a[i],a[min_index] = a[min_index],a[i]
    end
    a
end

p selection [2,1,-3,4,17,-5,37,8]

File: sorting/easy/selection.js
a = [7,5,3,1]
console.log(a)

for(i=0;i<a.length-1;i++){
    let min = a[i]
    let min_index = i
    for(j=i+1;j<a.length;j++){
        if(a[j]<a[i]){
            min = a[j]
            min_index = j
        }
    }
    if(a[i]>a[min_index]) [a[i],a[min_index]] = [a[min_index],a[i]]
}

console.log(a)